display/man.rs

```diff
@@ -0,0 +1,445 @@
+///
+/// [Vec<u8>] STDOUT of called `mandoc(1)` formatter.
+///
+/// # Errors
+///
+/// [std::io::Error] if file failed to execute `mandoc(1)` formatter.
+fn mandoc_format(man_page: &[u8], width: Option<u16>) -> Result<Vec<u8>, io::Error> {
+    let mut args = vec![];
+    if let Some(width) = width {
+        args.push("-O".to_string());
+        args.push(format!("width={width}"));
+    }
+
+    spawn("mandoc", &args, Some(man_page), Stdio::piped()).map(|output| output.stdout)
+}
+
+/// Formats man page content into appropriate format.
+///
+/// # Arguments
+///
+/// `man_page` - [Vec<u8>] with content that needs to be formatted.
+///
+/// # Returns
+///
+/// [Vec<u8>] STDOUT of called formatter.
+///
+/// # Errors
+///
+/// [ManError] if failed to execute formatter.
+fn format_man_page(man_page: Vec<u8>) -> Result<Vec<u8>, ManError> {
+    let width = get_page_width()?;
+
+    let formatters = [groff_format, nroff_format, mandoc_format];
+
+    for formatter in &formatters {
+        match formatter(&man_page, width) {
+            Ok(formatted_man_page) => return Ok(formatted_man_page),
+            Err(err) if err.kind() == io::ErrorKind::NotFound => continue,
+            Err(err) => return Err(err.into()),
+        }
+    }
+
+    Err(ManError(
+        "neither groff(1), nor nroff(1), nor mandoc(1) are installed".to_string(),
+    ))
+}
+
+/// Formats man page content into appropriate format.
+///
+/// # Arguments
+///
+/// `man_page` - [Vec<u8>] with content that needs to displayed.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [std::io::Error] if failed to execute pager or failed write to its STDIN.
+fn display_pager(man_page: Vec<u8>) -> Result<(), io::Error> {
+    let pager = std::env::var("PAGER").unwrap_or_else(|_| "more".to_string());
+
+    let args = if pager.ends_with("more") {
+        vec!["-s"]
+    } else {
+        vec![]
+    };
+
+    spawn(&pager, args, Some(&man_page), Stdio::inherit())?;
+
+    Ok(())
+}
+
+/// Displays man page
+///
+/// # Arguments
+///
+/// `name` - [str] name of system documentation.
+///
+/// # Returns
+///
+/// Nothing.
+///
+/// # Errors
+///
+/// [ManError] if man page not found, or any display error happened.
+fn display_man_page(name: &str) -> Result<(), ManError> {
+    let cat_output = get_man_page(name)?;
+    let formatter_output = format_man_page(cat_output)?;
+    display_pager(formatter_output)?;
+
+    Ok(())
+}
+
+/// Displays man page summaries for the given keyword.
+///
+/// # Arguments
+///
+/// `keyword` - [str] name of keyword.
+///
+/// # Returns
+///
+/// Nothing
+///
+/// # Errors
+///
+/// [ManError] if call of `apropros` utility failed.
+fn display_summary_database(keyword: &str) -> Result<(), ManError> {
+    let output: Output = Command::new("apropos").arg(keyword).output()?;
+
+    if !output.status.success() {
+        return Err(ManError("apropos command failed".to_string()));
+    }
+
+    let result = String::from_utf8_lossy(&output.stdout);
```

Comment:
Redundant buffer. If this was done to hide a possible error from `apropos` - this is unhelpful. Just use `spawn` instead of `output` [so the stdout is inherited](https://doc.rust-lang.org/std/process/struct.Command.html#method.stdout)